File: algorithm/src/main/java/baekjoon/p09094/Main.java
package baekjoon.p09094;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0) {
            System.out.println(solve(sc.nextInt(), sc.nextInt()));
        }
    }

    private static int solve(int n, int m) {
        int ret = 0;

        for (int a = 1; a < n; a++) {
            for (int b = a + 1; b < n; b++) {
                if ((a * a + b * b + m) % (a * b) == 0) {
                    ret++;
                }
            }
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p09094/Main.kt
package baekjoon.p09094

fun main(args: Array<String>) {
    repeat(readLine()!!.toInt()) {
        val inputs = readLine()!!.split(" ").map { it.toInt() }
        println(solve(inputs[0], inputs[1]))
    }
}

private fun solve(n: Int, m: Int): Int {
    var ret = 0

    for (a in 1 until n) {
        for (b in a + 1 until n) {
            if ((a * a + b * b + m) % (a * b) == 0) {
                ret++
            }
        }
    }

    return ret
}